src/components/interaction-designer/resource-editors/ResourceEditor.vue

```diff
@@ -70,33 +109,97 @@
 
   @Component({
     props: {
-        flow: {
-            type: Object as () => IFlow,
-            default: null,
-        },
-        label: {
-            type: [String, Number],
-        },
-        resource: {
-            type: Object as () => IResourceDefinition,
-            default: null,
-        }
+      block: {
+        type: Object as () => IBlock,
+        required: true,
+      },
+      flow: {
+        type: Object as () => IFlow,
+        default: null,
+      },
+      label: {
+        type: [String, Number],
+      },
+      resource: {
+        type: Object as () => IResourceDefinition,
+        default: null,
+      }
     },
 
-    mixins: [lang],
+    mixins: [
+      lang,
+      FlowUploader,
+      Permissions,
+      Routes,
+    ],
 
     components: {
       AudioLibrarySelector,
       ResourceVariantTextEditor,
+      UploadMonitor,
     },
   })
   export class ResourceEditor extends Vue {
     discoverContentTypesFor = discoverContentTypesFor
     findOrGenerateStubbedVariantOn = findOrGenerateStubbedVariantOn
+    findResourceVariantOverModesOn = findResourceVariantOverModesOn
     SupportedMode = SupportedMode
     SupportedContentType = SupportedContentType
 
+    handleFilesSubmittedFor(key, {data}) {
+      console.debug(`call handleFilesSubmittedFor`)
+      this.$store.dispatch('multimediaUpload/uploadFiles', {...data, key})
+    }
+
+    handleFileSuccessFor(key, langId, event) {
+      const {data: {file, json}} = event
+      const {uuid: jsKey} = this.block
+      const {
+            audio_file_id: id,
+            audio_uuid: filename,
+            created_at: {date: created_at},
+            description,
+            duration_seconds,
+          } = JSON.parse(json)
+      const extension = description.split('.')[description.split('.').length - 1]
+      const uploadedAudio: IAudioFile = {
+        id,
+        filename,
+        description,
+        language_id: langId,
+        duration_seconds,
+        original_extension: extension,
+        created_at
+      }
+
+      this.resource_setOrCreateValueModeSpecific({
+        resourceId: this.resource.uuid,
+        filter: {languageId: langId, contentType: SupportedContentType.AUDIO, modes: [SupportedMode.IVR]},
+        value: `https://www.viamo.io/audiofiles/play/${filename}/${extension}`
+      })
+      event.target.blur() // remove the focus from the `upload` Tab
+      this.pushAudioIntoLibrary(uploadedAudio)
```

Comment:
Can you explain why we need to edit to`updateAudioFileFor( )` under trees/stores/trees.js (according to tech AC) ? Or did I misunderstand it ?
With this current implementation, we're able to push the audio correctly

Reply:
According to Brett, it's ok without updateAudioFileFor( )